exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/__init__.py

```diff
@@ -222,10 +219,14 @@ def _collector_http_client(self) -> Optional[Collector]:
         return self._collector
 
     def export(self, spans) -> SpanExportResult:
+
+        tracer_provider = trace.get_tracer_provider()
+        resource = tracer_provider.resource
```

Comment:
if the Traceprovider from sdk is used, it defaults to Resource.create({}), which sets the default value to "unknown_serivce"
If the user is not setting a tracerprovider , then it defaults to DefaultTracerProvider  which is a no-op.
DefaultTracerProvider  doesn't have the resource attribute.

Reply:
I don't think this piece of code should rely on the knowledge you shared and avoid a safety check. TracerProvider behavior may change at any point and there is no way for this place to get notified. I'd suggest to add safety checks for at least resource being None.

Reply:
sure @owais. I will set the service_name to "unknown_service", if resource is None.

Reply:
I'm not sure if `unkown_service` should be used or if it should be left empty. I think it should be left empty but better to refer the spec and other otel projects and see what they do. 

That said, my point was about guarding against possible attribute errors if tracer_provider or trace_provider.resource are equal to None, not about setting a default value for service name.

Reply:
Yes @owais i agree with you. 
The spec says the SDK should set it to unknown_service.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#semantic-attributes-with-sdk-provided-default-value

The Resource.create({}) does exactly this. it checks for env variables and also attributes provided by the user. if it doesn't find any of these, it sets it to unkown_service.

Just to add incase of jaeger, jaeger.Process() accepts service_name as an argument. I'm not sure how jaeger works though.

@lonewolf3739 any other thoughts ?

Reply:
It is safe to check if provider has valid Resource instance before accessing `SERVICE_NAME` directly. There has been some discussion happening here https://github.com/open-telemetry/opentelemetry-python/issues/1522 about having multiple tracer providers. We don't say anywhere about the support of multiple tracer providers. If multiple tracer providers are used then this approach is no longer correct as `get_tracer_provider` returns global tracer provider and each tracer provider can have differ resource data.

Reply:
Wait, this is incorrect then I suppose because _I think_ people can use multiple tracer providers today. I don't think we have anything in place to prevent it from happening. If we need service_name in exporter, why not take it from span's resource?  I think that'd be more correct way to do it.

I know we are discussing removing span from resource but in that we'd come up with an alternative like passing resource from tracer to the rest of the pipeline (tracer > processor > exporter) or we'd not remove it at all. I'm fairly confident Otel as a project supports multiple tracer providers.

Reply:
To me, this looks like more of a translation issue. Otel's resource roughly translates to jaeger's process, at least as far as `service.name` is concerned. Perhaps we should update jaeger exporter to work with empty service name and then update translator to just copy Resource's attributes to Jaeger's process tags.

Reply:
from what I see,  currently when I try to set_tracer_provider() multiple times, it throws a warning and returns.
```
def set_tracer_provider(tracer_provider: TracerProvider) -> None:
    """Sets the current global :class:`~.TracerProvider` object.

    This can only be done once, a warning will be logged if any furter attempt
    is made.
    """
    global _TRACER_PROVIDER  # pylint: disable=global-statement

    if _TRACER_PROVIDER is not None:
        logger.warning("Overriding of current TracerProvider is not allowed")
        return

    _TRACER_PROVIDER = tracer_provider
```

Having said that, yes, this method will not work if a user can have multiple tracer_providers.

Reply:
@owais Yes, this is indeed incorrect in that case. I am not sure if everything would just work with multiple tracer providers. Here is what we have done in Zipkin exporter.
https://github.com/open-telemetry/opentelemetry-python/blob/03c6b7391d16236ad5d4e48b0cb0bff786c2e583/exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/node_endpoint.py#L46-L53

So far the (probably wrong) assumption is that we only support one global tracer provider. I think we can just use the resource from span for now and revisit if that changes in future?

Reply:
Yes, we definitely only support one global provider but they can be multiple other providers being used but not registered as the global provider.

Reply:
If we look at how the collector handles translation between formats, it translates OTLP/OC resource to Jaeger process and vice-versa. I don't see why we wouldn't do the same instead of overwriting possible values from span.Resource with global tracer provider's resource. I feel that'd be the correct thing to do but happy to hear other thoughts.

Reply:
Makes sense. I didn't know earlier about the use case you shared in the #1522. I thought using multiple tracer providers is not guaranteed to work and there is nothing that says we support it. This brings up another question for me. Should the SDK env config only be applied for global tracer provider or be applied for all the tracer providers or None at all? Let's take `TRACES_SAMPLER` for example there is no way users can configure it to use different sampling strategies for different providers with env? And they probably  don't want all of them to have same?

Reply:
> I thought using multiple tracer providers is not guaranteed to work and there is nothing that says we support it. 

It must work.

> Notwithstanding any global TracerProvider, some applications may want to or have to use multiple TracerProvider instances, e.g. to have different configuration (like SpanProcessors) for each (and consequently for the Tracers obtained from them), or because its easier with dependency injection frameworks. Thus, implementations of TracerProvider SHOULD allow creating an arbitrary number of TracerProvider instances.

I thought using multiple tracer providers is not guaranteed to work and there is nothing that says we support it.

Reply:
>  Should the SDK env config only be applied for global tracer provider or be applied for all the tracer providers or None at all? 

I think they should apply to every instance but explicitly passed config to initializers should take precedence. 

> Let's take TRACES_SAMPLER for example there is no way users can configure it to use different sampling strategies for different providers with env?

Sure, but all use cases where people would want multiple providers would almost always involve instantiating those providers with code so users can just pass explicit config. If our implementations don't accept explicit config through code then we should update them to support it.

Reply:
In any case, I think we are getting distracted from the issue at hand here. We can open an issue and discuss how to best support multiple tracer providers to consolidate all the discussion there and unblock this PR.